### DISCUSSION TITLE
Microsoft.Activities.Extensions Source And Publication

### DISCUSSION THREAD

**Discussion by TheBrambleShark** [dotnet/runtime#94575] [Q&A]
Hi all,

I'm not sure if this is the right place to ask this or even who to contact here. A while back, UiPath adapted Windows Workflow Foundation to target .NET 5 with their open source [CoreWF](https://github.com/UiPath/CoreWF/) library.

Back in the old NetFx days, Microsoft released the Microsoft.Activities.Extensions library, which contains a bunch of extensions and activities built for Windows Workflow Foundation. I have a WWF project that I am in the process of converting to CoreWF, but I am in need of this extension library.

I would like to do something similar as UiPath - open source this library and release compiled version for .NET 5 and newer, ensuring compatibility with the latest versions of .NET and CoreWF. The maintainers for the Microsoft.Activities.Extensions nuget package listing are Ron Jacobs and Jurgen Willis, but both of these individuals seem to have moved up in the Microsoft organization since developing these extensions.

My question is what is the correct chain to go through to see about open sourcing this code and allowing it to be published as open source? I want to make sure that everything is done above board.

Edit: Here's the codeplex page for this library: https://web.archive.org/web/20150914213844mp_/http://wf.codeplex.com/wikipage?title=Microsoft.Activities%20Overview&referringTitle=Home

**Comment by skypaint-96** [comment]
The Microsoft.Activities namespace is still something useful for sharepoint onprem workflows.
